src/Microsoft.Identity.Web.DownstreamApi/DownstreamApi.cs

```diff
@@ -265,18 +271,34 @@ internal partial class DownstreamApi : IDownstreamApi
                 return default;
             }
 
+            if (typeof(TOutput).IsAssignableFrom(typeof(HttpContent)))
+            {
+                return content as TOutput;
+            }
+
+            string? mediaType = content.Headers.ContentType?.MediaType;
+
             if (effectiveOptions.Deserializer != null)
             {
                 return effectiveOptions.Deserializer(content) as TOutput;
             }
             else
             {
                 string stringContent = await content.ReadAsStringAsync();
-                return JsonSerializer.Deserialize<TOutput>(stringContent, new JsonSerializerOptions { PropertyNameCaseInsensitive = true });
+                if (mediaType == "application/json")
+                {
+                    return JsonSerializer.Deserialize<TOutput>(stringContent, new JsonSerializerOptions { PropertyNameCaseInsensitive = true });
```

Comment:
this is fine because this assembly isn't trimmable I guess?

Reply:
do we have an issue to get this assembly trimmable?

Reply:
So, yes, there are trim warnings.
I wonder if we should, like in the Hybrid cache design, propose interfaces for serializers to implement instead of these delegates.
This would be a breaking change in Abstractions and IdWeb.

Reply:
Agreed, it is needed as part of the effort to get zero trim warnings

Reply:
See 3 [here](https://identitydivision.visualstudio.com/DevEx/_build/results?buildId=1290201&view=logs&j=096205eb-4599-56da-affa-13246cc49eac&t=02f496e3-4c4e-5ed1-c7f7-e313ca115ee1&l=39)

For some other ideas

Reply:
will address in a separate PR